packages/headless/src/controllers/result-list/headless-result-list.ts

```diff
@@ -51,6 +51,31 @@ export function buildResultList(
     dispatch(registerFieldsToInclude(options.fieldsToInclude));
   }
 
+  let lastFetchCompleted = 0;
+  let consecutiveFetches = 0;
+  const maxConsecutiveFetches = 5;
+  const maxDelayBetweenFetches = 200;
+
+  const triggerFetchMoreResult = () => {
+    if (engine.state.search.isLoading) {
+      return;
+    }
+
+    const delayBetweenFetches = Date.now() - lastFetchCompleted;
+    if (delayBetweenFetches < maxDelayBetweenFetches) {
+      consecutiveFetches++;
+      if (consecutiveFetches >= maxConsecutiveFetches) {
+        return engine.logger.error(
+          `The result list method "fetchMoreResults" execution prevented because it has been triggered consecutively ${maxConsecutiveFetches} times, with little delay. Please verify the conditions under which the function is called.`
+        );
+      }
+    } else {
+      consecutiveFetches = 0;
+    }
+
+    dispatch(fetchMoreResults()).then(() => (lastFetchCompleted = Date.now()));
+  };
+
```

Comment:
Open question: should the mechanism be embedded in the `fetchMoreResults` action instead of the ResultList controller?

Do we want the mechanism to be by-passed by calling the `fetchMoreResults` action directly? Why? or why not?
Scenario: multiple result list controller instances dispatching the action at the same time.

Reply:
I'm not sure that scenario is realistic, if you dispatch a search while another one is still going, it's going to abort the first, log an error, and run the second. That was the first PR for this story. 2 result lists dispatching different requests would require 2 engines to be successful.

This current PR is directed towards an infinite scroll scenario, which is done through the controller, usually. If a user would like to use the dispatch directly, then he would have more control, but also more responsibility 😃  like for the rest of the actions. I could maybe be done at the action level, but IMO it adds a bit more complexity which might not be necessary.